File: problems/Array/1684.php
<?php

class Solution
{
    function countConsistentStrings(string $allowed, array $words): int
    {
        $count = 0;
        foreach ($words as $word) {
            $isLegal = true;
            for ($i = 0; $i < strlen($word); $i++) {
                if (!str_contains($allowed, $word[$i])) {
                    $isLegal = false;
                    break;
                }
            }
            if ($isLegal) {
                $count++;
            }
        }

        return $count;
    }
}

$solution = new Solution();
print_r($solution->countConsistentStrings("ab", ["ad", "bd", "aaab", "baa", "badab"]));

File: problems/String/1684.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: list[str]) -> int:
        invalid = 0

        for word in words:
            for w in word:
                if w not in allowed:
                    invalid += 1
                    break

        return len(words) - invalid

if __name__ == '__main__':
    print(Solution().countConsistentStrings("ab", ["ad","bd","aaab","baa","badab"]))
